sdk/nodejs/runtime/closure/createClosure.ts

```diff
@@ -1220,23 +1317,76 @@ function findNormalizedModuleName(obj: any): string | undefined {
     return undefined;
 }
 
-const legalNameRegex = /^[a-zA-Z_][0-9a-zA-Z_]*$/;
-export function isLegalMemberName(n: string) {
-    return legalNameRegex.test(n);
+async function hasTruthyMemberAsync(obj: any, memberName: string): Promise<boolean> {
+    if (!obj) {
+        return false;
+    }
+
+    return obj[memberName] ? true : false;
 }
 
-export function isLegalFunctionName(n: string) {
-    if (!isLegalMemberName(n)) {
-        return false;
+function createClosurePropertyDescriptor(
+    nameOrSymbol: string | symbol, descriptor: PropertyDescriptor): ClosurePropertyDescriptor {
+
+    if (nameOrSymbol === undefined) {
+        throw new Error("Was not given a name or symbol");
     }
 
-    const scanner = ts.createScanner(
-        ts.ScriptTarget.Latest, /*skipTrivia:*/false, ts.LanguageVariant.Standard, n);
-    const tokenKind = scanner.scan();
-    if (tokenKind !== ts.SyntaxKind.Identifier &&
-        tokenKind !== ts.SyntaxKind.ConstructorKeyword) {
-        return false;
+    const copy: ClosurePropertyDescriptor = { ...descriptor };
```

Comment:
I got burned once by this doing a shallow copy and not a deep copy. Does this depend on the `value` field getting deep copied?

Reply:
we should be ok there.  this is really just because i don't wnat to reuse their top level descriptor object, but i want to ensure it's in the form we want.  The value doesn't need to be deeply cloned.  indeed, we don't want it to be so that we preserve object identity (critical for our algorithm terminating loops).